spamc.go

```diff
@@ -67,10 +68,148 @@ func (c *Client) send(
 	return conn, nil
 }
 
+// rearrangeMessageReader will rearrange the message reader to make sure SA will understand the message
+func (c Client) rearrangeMessageReader(message io.ReadSeeker, size int64) (io.Reader, int64, error) {
+	var headers bytes.Buffer
+	var rest bytes.Buffer
+
+	isASCII := func(m []byte) bool {
+		for _, c := range m {
+			if c > 127 {
+				return false
+			}
+		}
+		return true
+	}
+
+	r := bufio.NewReader(message)
+
+	var returnError error
```

Comment:
Is there a reason in particular you only `return` at the end of the function, or can we `return` earlier?

Reply:
It is not really feasible to return early (see the body termination check and constructing the io.MultiReader before return).  We can return early if we want to error out and not send anything to SA if there is an error but this would have a far reaching impact as sending 'Message' would error out on 'malformed MIME header line:'